docs/adr/0004-framework.md

```diff
@@ -0,0 +1,131 @@
+- Missing out on a bunch of automatic optimisations Next.js does, and going a bit "against the grain" on the main use case the Next.js folks focus on.
+
+#### 2. Server-side rendering (SSR)
+
+This involves having Next.js run our back-end, where it can tailor the initially-rendered DOM to the current user.
+
+Pros:
+- Quick rendering of the first response to the user.
+- Back-end wouldn't need to add an API endpoint for all the data we currently use.
+- Can benefit from more automatic optimisations provided by Next.js, e.g. [image optimisation](https://nextjs.org/docs/pages/building-your-application/optimizing/images) to speed up data transfer and minimise cumulative layout shift.
+
+Cons:
+- Front-end coupled to the back-end, so preview deployments still aren't a given.
+- Has significant architectural impacts on the backend.
+- Not clear yet how e.g. FxA integration and cron jobs would work.
+
+#### 3. Server-side rendering in front of our current back-end (hybrid SSR?)
+
+Pros:
+- We can potentially leave more of our current back-end code unmodified in the short term.
+- Quick rendering of the first response to the user.
+- Can benefit from more automatic optimisations provided by Next.js, e.g. [image optimisation](https://nextjs.org/docs/pages/building-your-application/optimizing/images) to speed up data transfer and minimise cumulative layout shift.
+
+Cons:
+- Not clear how incremental an eventual migration to plain SSR would be - i.e. migrating a whole endpoint at a time is still a fairly big chunk, and I'm not sure if that granularity will even be feasible.
+- Might involve lots of complexity to e.g. share sessions between the two back-ends.
+- Not sure what impact this will have w.r.t. Kubernetes (which I know basically nothing about).
+- Still has significant architectural impacts on the backend.
+
+
+
+### Leave as-is
+
+* Good, the current website has gone through extensive testing, fox fooding and is working
+* Good, Little Third-Party dependencies
+* Good, No throw away code
+* Good, Conveys our support for standards and the web standards process
+* Bad, Front-end testing requires additional set-up (using a full browser) which would impact test speeds and contribute to flakiness.
+* Bad, No Virtual DOM: Current architecture may be slower than React, as it lacks the optimizations provided by a virtual DOM.
+* Bad, Harder to learn our snowflake framework (e.g. our use of a proxy on one page to update the DOM)
+* Bad, Easy for server and client-side state to get out of sync
+* Bad, Hard to re-use code on the client-side (see: dynamic Fluent strings)
+* Bad, Hard to statically analyse, and in general doesn't make use of a lot of tooling that can help us deliver better quality code without us manually set up
+* Bad, Poor front-end developer experience: Writing in template literals removes tag coloring, auto-complete functions, etc.
```

Comment:
This is a real, daily paper cut. 

_If there's a solve now anyone has, please share it. I've heard from other devs about this and experienced it first hand._ 

HTML inside JavaScript as a string: 👎 
<img width="638" alt="image" src="https://user-images.githubusercontent.com/2692333/236512360-7938d209-4aec-4030-b5b2-f376cd5e03d2.png">

Reply:
Hmm, there's this tool called [Handlebars](https://handlebarsjs.com/) that has widespread editor support? :P

Reply:
😭 😭 😭

Reply:
@maxxcrawford I use the Template Literals VSCode extension: https://marketplace.visualstudio.com/items?itemName=julienetie.vscode-template-literals

Works very well for me for template literal code, but I sometimes need to disable it when working in non-template JS. I had been meaning to add this extension to the recommended tools section of ReadMe, but never got around to it.

Reply:
@maxxcrawford there might be other syntax highlight extensions as well, please share if you find a better one!  Regardless, I hope the one above alleviates the added "con" of no syntax highlight.